<launch>

  <!-- Bag -->
  <node name="rosbag" 
        pkg="rosbag" 
        type="play" 
        respawn="false"
        args="-s 1 -r 1 /home/sst/work/data/dataset/euroc_sequences/V1_01_easy.bag">
  </node> 

  <!-- Visual-inertial Estimator -->
  <node pkg="edge_imu_bias"
        name="edge_imu_bias"
        type="slam"       
        output="screen">
    <!--remap from="~image"    to="/camera/abc"/-->
    <param name="intrinsics_calibration_file"		value="$(find edge_imu_bias)/calib/combine2.yml"/>
    <param name="extrinsics_calibration_file"		value="$(find edge_imu_bias)/calib/visensor.yml"/>
    <param name="skipFrameNum"				value= "1"/>
    <param name="onUAV"					value= "false"/>
    <param name="enable_histogramEqualization"		value= "false"/>
    <param name="errorTranslationThreshold"		value= "0.2"/>
    <param name="errorAngleThreshold"			value= "3"/>
    <param name="adaptiveCannyThreshold"		value= "true"/>
    <param name="enable_LoopClosure"			value= "true"/>
    <param name="frontMarginalization"			value= "true"/>
    <param name="loopClosureInterval"			value= "0"/>
    <param name="visualWeight"				value= "100"/>
    <param name="bias_g_x_0"				value= "-0.002295"/>
    <param name="bias_g_y_0"				value= "0.024939"/>
    <param name="bias_g_z_0"				value= "0.081667"/>
    <param name="bias_a_x_0"				value= "-0.023601"/>
    <param name="bias_a_y_0"				value= "0.121044"/>
    <param name="bias_a_z_0"				value= "0.074783"/>
    <param name="denseOrNot"				value= "true"/>
    <param name="edgeIterationNum"			value= "10"/>
    <param name="huber_r_v"			value= "0.5"/>
    <param name="huber_r_w"			value= "0.5"/>
    <param name="IMUorNot"			value= "true"/>

        <param name="max_feature_cnt" type="int" value="200" />
        <param name="num_of_iter" type="int" value="10" />
        <param name="calib_threshold_ric" type="double" value="0.0" />
        <param name="calib_threshold_tic" type="double" value="1.0" />
        <param name="init_depth" type="double" value="15.0" />
        <param name="gradient_threshold" type="double" value="1.0" />
        <param name="feature_threshold" type="double" value="5.0" />
        <param name="min_parallax" type="double" value="30.0" />
        <param name="min_parallax_point" type="double" value="5.0" />
        <param name="error_threshold" type="double" value="10000000000.0" />
        <param name="show_histogram" type="bool" value="false" />
        <param name="show_graph" type="bool" value="false" />
        <param name="show_html" type="bool" value="false" />
        <param name="multi_thread" type="bool" value="false" />
        <param name="imu_rate" type="double" value="200" />
        <param name="bias_acc_threshold" type="double" value="0.0005" />
        <param name="bias_gyr_threshold" type="double" value="0.0003" />
        <param name="solver_time" type="double" value="0.100" />
        <param name="compensate_rotation" type="bool" value="true" />
        <param name="acc_n" type="double" value="8.0e-3" />
        <param name="acc_w" type="double" value="4.0e-5" />
        <param name="gyr_n" type="double" value="12.0e-4" />
        <param name="gyr_w" type="double" value="4.0e-6" />
  </node>  

</launch>
